source/Calamari.AzureWebApp.Tests/DeployAzureWebCommandFixture.cs

```diff
@@ -0,0 +1,395 @@
+
+            var pathToPackage = Path.Combine(tempPath.DirectoryPath, CreateNugetPackage("Hello", tempPath.DirectoryPath));
+
+            await CommandTestBuilder.CreateAsync<DeployAzureWebCommand, Program>()
+                .WithArrange(context =>
+                {
+                    AddDefaults(context, webAppName);
+                    context.WithPackage(pathToPackage);
+                })
+                .Execute();
+
+            await AssertContent(webApp.DefaultHostName, actualText);
+        }
+
+        [Test]
+        public async Task Deploy_WebApp_With_PhysicalPath()
+        {
+            var webAppName = SdkContext.RandomResourceName(nameof(DeployAzureWebCommandFixture), 60);
+
+            var webApp = await CreateWebApp(webAppName);
+
+            using var tempPath = TemporaryDirectory.Create();
+            var actualText = "Hello World";
+
+            File.WriteAllText(Path.Combine(tempPath.DirectoryPath, "index.html"), actualText);
+            const string rootPath = "Hello";
+
+            await CommandTestBuilder.CreateAsync<DeployAzureWebCommand, Program>()
+                .WithArrange(context =>
+                {
+                    AddDefaults(context, webAppName);
+                    context.Variables.Add(SpecialVariables.Action.Azure.PhysicalPath, rootPath);
+
+                    context.WithFilesToCopy(tempPath.DirectoryPath);
+                })
+                .Execute();
+
+            await AssertContent(webApp.DefaultHostName, actualText, rootPath);
+        }
+
+        Task<IWebApp> CreateWebApp(string webAppName)
+        {
+            return azure.WebApps
+                .Define(webAppName)
+                .WithExistingWindowsPlan(appServicePlan)
+                .WithExistingResourceGroup(resourceGroup)
+                .WithRuntimeStack(WebAppRuntimeStack.NETCore)
+                .CreateAsync();
+        }
+
+        async Task AssertContent(string hostName, string actualText, string rootPath = null)
+        {
+            var result= await client.GetStringAsync($"https://{hostName}/{rootPath}");
+
+            result.Should().Be(actualText);
+        }
+
+        void AddDefaults(CommandTestBuilderContext context, string webAppName)
+        {
+            context.Variables.Add(AzureAccountVariables.SubscriptionId,
+                subscriptionId);
+            context.Variables.Add(AzureAccountVariables.TenantId,
+                tenantId);
+            context.Variables.Add(AzureAccountVariables.ClientId,
+                clientId);
+            context.Variables.Add(AzureAccountVariables.Password,
+                clientSecret);
+            context.Variables.Add(SpecialVariables.Action.Azure.WebAppName, webAppName);
+            context.Variables.Add(SpecialVariables.Action.Azure.ResourceGroupName, resourceGroup.Name);
+        }
+
+        static string CreateNugetPackage(string packageId, string filePath)
```

Comment:
This can be moved to some common test class

Reply:
done